### PR TITLE
Add specific chip to Marshal

### PR DESCRIPTION
**Pull request by Ampersandrew**
Added an application to allocate Marshal chips.

Converted the Player allocation application and the Marshal allocation applications to app v2.

Moved the chip allocation reporting out of Actor into Chips, since it is used for both Player and Marshal reports.

### CODE REVIEW

File: module/appsv2/allocate-chips-marshal.mjs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  /** @inheritdoc */
+  async _prepareContext(options = {}) {
+    let context = await super._prepareContext(options);
+
+    // Get the actors who can hold chips
+    const chipBearers = game.actors.filter((a) => a.system.hasChips === true);
+
+    const posse = [];
+
+    // eslint-disable-next-line no-restricted-syntax
```
Comment (TheSpyder) [inline module/appsv2/allocate-chips-marshal.mjs:44]: hmm we need to adjust our eslint rules if it's complaining about stuff like that

File: module/init/socket-functions.mjs [hunk 2]
```diff
@@ -129,6 +129,45 @@ async function socketAddChipsActor(actorId, chips) {
...
+  blue = typeof blue === 'number' ? Math.floor(blue) : 0;
+
+  let total = white + red + blue;
+
+  // If we aren't actually adding any chips
+  if (total <= 0) return;
+
+  const adding = { white, red, blue };
+  const chatStr = Chips.makeMarshalReport('Added', adding);
+  _chatMessage(ChatMessage.getSpeaker(), 'Marshal', chatStr);
```
Comment (TheSpyder) [inline module/init/socket-functions.mjs]: This broadcasts for all players to see, did you mean to post to the marshal only?

**Review by TheSpyder — APPROVED** [review]
The styles aren't working for me, the labels are blank, but the buttons all seem to operate correctly

Screenshot 2024-06-20 at 8 46 43 PM

File: templates/chip/allocator-marshal.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<section class="{{cssClass}} directory flexcol" id="{{cssId}}" data-tab="{{tabName}}">
+
+  <header class="chip-allocator-header">
+      <div class="flexrow">
+        <h3 class="chip-allocator noborder">{{localize 'DLC.app.ChipAllocator'}}</h3>
+      </div>
+  </header>
+
+  <div class="chip-row">
+    <label class="chip-label-character" for="white">White</label>
```
Comment (TheSpyder) [inline templates/chip/allocator-marshal.html:10]: `chip-label-character` defaults to text color `var(--color-text-dark-1)` on my system. The default allocator dialog has a texture background so the black text is visible

File: module/appsv2/allocate-chips-marshal.mjs (hunk 1 above)
Comment (Ampersandrew) [reply module/appsv2/allocate-chips-marshal.mjs:44]: I looked up the thing it's complaining about and it was suggested that it was maybe a good idea more than 10 years ago, but javascript has moved on. Happy to remove this rule, I just disable it when it complains anyway.

File: module/init/socket-functions.mjs (hunk 2 above)
Comment (Ampersandrew) [reply module/init/socket-functions.mjs]: I did, the Marshal can see his own chip pot and doesn't really need a report that he's done the thing he just did. We could possibly add a setting to turn it off.

File: templates/chip/allocator-marshal.html (hunk 3 above)
Comment (Ampersandrew) [reply templates/chip/allocator-marshal.html:10]: There are two themes in App v2 apparently, they've added a dark theme. My system is using the light theme, so I didn't see this.

File: module/init/socket-functions.mjs (hunk 2 above)
Comment (TheSpyder) [reply module/init/socket-functions.mjs]: yes if we can't get it constraining messages so only the marshal can see them we should just stop broadcasting this.

File: templates/chip/allocator-marshal.html (hunk 3 above)
Comment (TheSpyder) [reply templates/chip/allocator-marshal.html:10]: right, yes, it's probably set to use the system theme and I have mac switching to dark mode at night.

**Pull request merged**